packages/nextjs/cypress/e2e/contract_interaction.cy.ts

```diff
@@ -0,0 +1,719 @@
+    "outputs": [
+      {
+        "internalType": "uint256",
+        "name": "",
+        "type": "uint256"
+      }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+  },
+  {
+    "inputs": [
+      {
+        "internalType": "address",
+        "name": "_validatorAddress",
+        "type": "address"
+      }
+    ],
+    "name": "validatorAdd",
+    "outputs": [],
+    "stateMutability": "nonpayable",
+    "type": "function"
+  },
+  {
+    "inputs": [],
+    "name": "validatorFee",
+    "outputs": [
+      {
+        "internalType": "uint256",
+        "name": "",
+        "type": "uint256"
+      }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+  },
+  {
+    "inputs": [
+      {
+        "internalType": "uint256",
+        "name": "",
+        "type": "uint256"
+      }
+    ],
+    "name": "validatorList",
+    "outputs": [
+      {
+        "internalType": "address",
+        "name": "",
+        "type": "address"
+      }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+  },
+  {
+    "inputs": [],
+    "name": "validatorListCount",
+    "outputs": [
+      {
+        "internalType": "uint256",
+        "name": "",
+        "type": "uint256"
+      }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+  },
+  {
+    "inputs": [
+      {
+        "internalType": "address",
+        "name": "_validatorAddress",
+        "type": "address"
+      }
+    ],
+    "name": "validatorRemove",
+    "outputs": [],
+    "stateMutability": "nonpayable",
+    "type": "function"
+  },
+  {
+    "inputs": [],
+    "name": "validatorTotalFee",
+    "outputs": [
+      {
+        "internalType": "uint256",
+        "name": "",
+        "type": "uint256"
+      }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+  },
+  {
+    "inputs": [
+      {
+        "internalType": "uint256",
+        "name": "_fee",
+        "type": "uint256"
+      }
+    ],
+    "name": "validatorsetFee",
+    "outputs": [],
+    "stateMutability": "nonpayable",
+    "type": "function"
+  }
+]`;
+
+describe("Contract Interaction", () => {
+  before(() => {
+    cy.wakeUpHeimdall();
+  });
+
+  it("should load DAI contract and interact with its balanceOf method", () => {
+    cy.visit("http://localhost:3000");
+    cy.loadContract("0x6B175474E89094C44Da98b954EedeAC495271d0F", 1000);
+    cy.url().should("include", "/0x6B175474E89094C44Da98b954EedeAC495271d0F/1");
+    cy.interactWithMethod("balanceOf", "0x6B175474E89094C44Da98b954EedeAC495271d0F");
+  });
+
+  it("should load proxy contract on Base and interact with its balanceOf method", () => {
+    cy.visit("http://localhost:3000");
+    cy.selectNetwork("Base");
+    cy.loadContract("0xca808b3eada02d53073e129b25f74b31d8647ae0", 1000);
+    cy.url().should("include", "/0xca808b3eada02d53073e129b25f74b31d8647ae0/8453");
+    cy.contains("Implementation Address").should("be.visible");
+    cy.wait(1000); // wait for: the method card to re-render
```

Comment:
I think we don't need this wait right? Since in `cy.interactiWithMethod` we have : 

```ts
cy.contains(methodName).click();
```

And it will by default retry for 4 sec until the method card is visible? as mentioned here https://github.com/BuidlGuidl/abi.ninja/pull/140/files#r1697899465

Reply:
The problem with this one is that when cypress clicks on the balanceOf method in the test for base:
1. The method card renders
2. Cypress starts typing the address "0xd
3. At this moment card gets re-rendered on the page
4. Cypress clicks the Read 📡 button with no address 
5. The test fails

How I think we can solve this:
This failure points that we have a problem with the rendering logic in the app. So, until we fix it in another PR, we should keep this delay of 1 second to make the tests pass.
After we fix the underlying issue, we can safely remove the delay.

Reply:
>3. At this moment card gets re-rendered on the page
> This failure points that we have a problem with the rendering logic in the app. So, until we fix it in another PR, we should keep this delay of 1 second to make the tests pass.

Ooohhh I see! yeah lets tackle it another PR